nixos/modules/misc/documentation.nix

```diff
@@ -51,7 +51,11 @@ let
             let wholeName = "${namePrefix}.${name}"; in
             if isAttrs value then
               scrubDerivations wholeName value
-              // (optionalAttrs (isDerivation value) { outPath = "\${${wholeName}}"; })
+              // optionalAttrs (isDerivation value) {
+                outPath = "\${${wholeName}}";
+                drvPath = wholeName;
+                __pretty = _: wholeName; val = null;
+              }
```

Comment:
I think I agree. My main concern are things like `default = "... ${pkgs.foo} ..."` without a `defaultText`. The `"\${${wholeName}}"` hack currently makes this render "nicely" but this is arguably a bug with `toPretty`: the interpolation should be escaped, so that it doesn't look nice anymore. I think I'd like to do something like

```
// optionalAttrs (isDerivation value) {
  outPath = throw "attempt to use a derivation in options documentation";
  drvPath = # same
}
```

It's a bit annoying that this has to be decided at the level of the evaluation, rather than in the generic `make-options-doc`, so that out-of-tree projects have to make their own decision. But probably fine.

Reply:
> The `"\${${wholeName}}"` hack currently makes this render "nicely" but this is arguably a bug with `toPretty`:

I agree that this is a bug, but is it a reason to drop easy `pkgs.foo` interpolation from the docs?

It could be made to work well by
 - fixing the escaping first
 - making the escaping customizable (function passed to `toPretty`)
 - pass a custom escaping function, which at the end replaces a special syntax like `$$${{{` to `${` so that antiquotations are representable without stealing too much from the normal string literals.
 - make `pkgs.foo.outPath = "\$\$\${{{pkgs.foo}"`
 - toPretty will print the right expression

Reply:
Considering that this feature is not used by anything in-tree (manual still builds with a `throw`), I think it's worth trying to move in the direction of least hackiness, which IMO is something like https://github.com/NixOS/nix/issues/1474.

Reply:
Alright. Then let's keep using the least hacky option available with the Nix we have today:

```nix
{
  defaultText = lib.literalExpression ''
    "''${pkgs.foo}/bin/foo"
  '';
}
```